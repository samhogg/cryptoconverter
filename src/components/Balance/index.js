import React from "react";
import PropTypes from "prop-types";
import { BalanceCurrencyPair } from "..";
import {
  BalanceContainer,
  BalanceListTitle,
  BalanceList
} from "./styledComponents";

const Balance = props => (
  <BalanceContainer>
    <BalanceListTitle>Account Balance</BalanceListTitle>
    <BalanceList>
      {Object.keys(props).map(ticker => (
        <BalanceCurrencyPair
          key={ticker}
          ticker={ticker}
          quantity={props[ticker]}
        />
      ))}
    </BalanceList>
  </BalanceContainer>
);

Balance.propTypes = {
  usd: PropTypes.number.isRequired,
  btc: PropTypes.number.isRequired
};

export default Balance;
