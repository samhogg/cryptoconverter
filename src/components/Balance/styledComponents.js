import styled from "styled-components";

const BalanceContainer = styled.section`
  padding: 1rem;
  border-bottom: 1px solid #ddd;
`;

const BalanceListTitle = styled.h2`
  margin: 0 0 1rem 0;
`;

const BalanceList = styled.ul`
  margin: 0;
  padding: 0;
  list-style: none;
  display: flex;
`;

export { BalanceContainer, BalanceListTitle, BalanceList };
