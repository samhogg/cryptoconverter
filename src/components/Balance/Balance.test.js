import React from "react";
import Balance from ".";
import renderer from "react-test-renderer";

it("Renders with zero balances", () => {
  const tree = renderer.create(<Balance usd={0} btc={0} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("Renders with non zero balances", () => {
  const tree = renderer.create(<Balance usd={100} btc={1000} />).toJSON();
  expect(tree).toMatchSnapshot();
});
