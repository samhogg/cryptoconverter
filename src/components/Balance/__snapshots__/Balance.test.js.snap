// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders with non zero balances 1`] = `
<section
  className="styledComponents__BalanceContainer-cIYnGy iIjgYz"
>
  <h2
    className="styledComponents__BalanceListTitle-bPaUMl hDojKG"
  >
    Account Balance
  </h2>
  <ul
    className="styledComponents__BalanceList-bqFABO eHXqVh"
  >
    <li
      className="styledComponents__BalanceItem-fDtCqf gzMZya"
    >
      <div
        className="styledComponents__BalanceTicker-dDjLel bpTZSY"
      >
        usd
      </div>
      <div
        className="styledComponents__BalanceQuantity-kFtsKU jMHtFP"
      >
        <span
          className="styledComponents__BalancePrefix-fCshfk eIVmaR"
        >
          $
        </span>
        100.00
      </div>
    </li>
    <li
      className="styledComponents__BalanceItem-fDtCqf gzMZya"
    >
      <div
        className="styledComponents__BalanceTicker-dDjLel bpTZSY"
      >
        btc
      </div>
      <div
        className="styledComponents__BalanceQuantity-kFtsKU jMHtFP"
      >
        <span
          className="styledComponents__BalancePrefix-fCshfk eIVmaR"
        >
          ₿
        </span>
        1,000.000000
      </div>
    </li>
  </ul>
</section>
`;

exports[`Renders with zero balances 1`] = `
<section
  className="styledComponents__BalanceContainer-cIYnGy iIjgYz"
>
  <h2
    className="styledComponents__BalanceListTitle-bPaUMl hDojKG"
  >
    Account Balance
  </h2>
  <ul
    className="styledComponents__BalanceList-bqFABO eHXqVh"
  >
    <li
      className="styledComponents__BalanceItem-fDtCqf gzMZya"
    >
      <div
        className="styledComponents__BalanceTicker-dDjLel bpTZSY"
      >
        usd
      </div>
      <div
        className="styledComponents__BalanceQuantity-kFtsKU jMHtFP"
      >
        <span
          className="styledComponents__BalancePrefix-fCshfk eIVmaR"
        >
          $
        </span>
        0.00
      </div>
    </li>
    <li
      className="styledComponents__BalanceItem-fDtCqf gzMZya"
    >
      <div
        className="styledComponents__BalanceTicker-dDjLel bpTZSY"
      >
        btc
      </div>
      <div
        className="styledComponents__BalanceQuantity-kFtsKU jMHtFP"
      >
        <span
          className="styledComponents__BalancePrefix-fCshfk eIVmaR"
        >
          ₿
        </span>
        0.000000
      </div>
    </li>
  </ul>
</section>
`;
